### DISCUSSION TITLE
Pydantic not being able to reference ObjectId of another model

### DISCUSSION THREAD

**Discussion by Sridatree** [pydantic/pydantic#5757] [Question] [answered]
Hello I'm new to Pydantic so apologize if this sounds stupid.
I have two collections, Customers and Account. Customer has a list of accounts. On each Account object, there is a customer identifier which is actually the mongo object id of the customer collection.
I have used json.ENCODERS_BY_TYPE[ObjectId]=str. The problem with this is the customer identifier is being stored as a string in the database where I want it to be stored as an object id.
How do I make pydantic do that?

Additionally I tried to use bson.ObjectId() to create a new ObjectId for customer identifier but it lands me to this error:
RuntimeError: error checking inheritance of ObjectId('object id generated') (type: ObjectId)

**Comment by me-v2** [comment, marked as answer]
use beanie orm based on pydantic .

**Reply by Sridatree** [reply]
Thank you so much!
